### PR TITLE
added printer_get_settings_from_options function

### PR DESCRIPTION
**Pull request by divyashk**

### CODE REVIEW

File: modules/printbackends/gtkprintbackendcpdb.c [hunk 1]
```diff
@@ -74,12 +74,57 @@ cpdb_get_printer_list(GtkPrintBackend *backend)
...
+
+    option = gtk_printer_option_set_lookup(options, "gtk-n-up");
+    if (option)
+      gtk_print_settings_set ( settings, GTK_PRINT_SETTINGS_NUMBER_UP, option->value );
+
+    option = gtk_printer_option_set_lookup( options, "gtk-n-up-layout");
+    if (option)
+      gtk_print_settings_set( settings, GTK_PRINT_SETTINGS_NUMBER_UP_LAYOUT, option->value);
+
+    ppd_file_t *ppd_file = gtk_print_cups_get_ppd(GTK_PRINTER_CUPS(printer));
```
Comment (rithvikp1998) [inline modules/printbackends/gtkprintbackendcpdb.c]: `gtk_print_cups_get_ppd` is a cups backend method. We should either have a similar method for CPDB or drop it for now and add a TODO and do it later (or even drop it altogether if it got deprecated)

File: modules/printbackends/gtkprintbackendcpdb.c [hunk 2]
```diff
@@ -74,12 +74,57 @@ cpdb_get_printer_list(GtkPrintBackend *backend)
...
+      print_at_time = gtk_print_settings_get (settings, "print-at-time");
+
+      if (strcmp (print_at, "at") == 0)
+      {
+        char *utc_time = NULL;
+        utc_time = localtime_to_utctime (print_at_time);
+
+        if (utc_time != NULL)
+        {
+          gtk_print_settings_set (settings, "cups-job-hold-until", utc_time);
```
Comment (rithvikp1998) [inline modules/printbackends/gtkprintbackendcpdb.c]: This is also a cups option. CPDB supports having multiple backends (CUPS, Google Cloud Print, File etc) in parallel, so we might not be using a CUPS printer always

File: modules/printbackends/gtkprintbackendcpdb.c [hunk 3]
```diff
@@ -74,12 +74,57 @@ cpdb_get_printer_list(GtkPrintBackend *backend)
...
+          g_free (utc_time);
+        }
+        else
+          gtk_print_settings_set (settings, "cups-job-hold-until", print_at_time);
+      }
+      else if (strcmp (print_at, "on-hold") == 0)
+        gtk_print_settings_set (settings, "cups-job-hold-until", "indefinite");
```
Comment (rithvikp1998) [inline modules/printbackends/gtkprintbackendcpdb.c]: Same here

File: modules/printbackends/gtkprintbackendcpdb.c [hunk 4]
```diff
@@ -74,12 +74,57 @@ cpdb_get_printer_list(GtkPrintBackend *backend)
...
+    if (option)
+      gtk_print_settings_set ( settings, GTK_PRINT_SETTINGS_NUMBER_UP, option->value );
+
+    option = gtk_printer_option_set_lookup( options, "gtk-n-up-layout");
+    if (option)
+      gtk_print_settings_set( settings, GTK_PRINT_SETTINGS_NUMBER_UP_LAYOUT, option->value);
+
+    ppd_file_t *ppd_file = gtk_print_cups_get_ppd(GTK_PRINTER_CUPS(printer));
+    if ( ppd_file != NULL)
+    {
```
Comment (rithvikp1998) [inline modules/printbackends/gtkprintbackendcpdb.c]: All the remaining options are only applicable if ppd file is present?

**Review by rithvikp1998 — CHANGES_REQUESTED** [review]

File: modules/printbackends/gtkprintbackendcpdb.c (hunk 4 above)
Comment (divyashk) [reply modules/printbackends/gtkprintbackendcpdb.c]: Yes, a similar check was performed in the cups backend

**Comment by divyashk** [comment]
@rithvikp1998 Please review the changes.

**Pull request merged**